### PR TITLE
Unlink when done (C and python bindings)

### PR DESCRIPTION
**Pull request by btovar**
## Proposed changes

Adds the declare_file VINE_UNLINK_WHEN_DONE  (unlink_when_done=True in python) to unlink a file at the manager's disk when the file refcount goes to 0.

This pr only implements the mechanism, but it has not been integrated to other parts of the code (e.g., PythonTask).

## Post-change actions

Put an 'x' in the boxes that describe post-change actions that you have done.
The more 'x' ticked, the faster your changes are accepted by maintainers.

- [x] `make test`       Run local tests prior to pushing.
- [x] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [x] `make lint`       Run lint on source code prior to pushing.
- [x] Manual Update     Did you update the manual to reflect your changes, if appropriate? This action should be done after your changes are approved but not merged.
- [x] Type Labels       Select github labels for the type of this change: bug, enhancement, etc.
- [x] Product Labels    Select github labels for the product affected: TaskVine, Makeflow, etc.
- [x] PR RTM            Mark your PR as ready to merge.

## Additional comments
This section is dedicated to changes that are ambitious or complex and require substantial discussions. Feel free to start the ball rolling.

### CODE REVIEW

File: doc/manuals/taskvine/index.md [hunk 1]
```diff
@@ -1199,6 +1199,56 @@ warranted:
...
 
+### Automatic Garbage Collection on Disk
+
+For workflows that generate intermediate files, TaskVine can automatically
+delete them from disk when the application indicates that they will not be
+needed anymore:
+
+=== "Python"
+    ```python
+    intemerdiate = m.declare_file("intermediate_file", unlink_when_done=True)
```
Comment (dthain) [inline doc/manuals/taskvine/index.md:1210]: spelling

**Review by dthain — COMMENTED** [review]
The code looks fine to me.  For the documentation, I think the difference between "intermediate files" (at the submit side) and "temporary files" (in the cluster) is going to be confusing.   Is there a term that would better clarify the distinction?

File: doc/manuals/taskvine/index.md (hunk 1 above)
Comment (btovar) [reply doc/manuals/taskvine/index.md:1210]: something like `partial_result = ...` ?

**Comment by dthain** [comment]
Yes, that looks better to me, but check spelling of intermediate in a few places.

**Pull request merged**